### DISCUSSION TITLE
Renovate can not read our org internal GitHub action tags

### DISCUSSION THREAD

**Discussion by zisuu** [renovatebot/renovate#31770] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, Renovate 38.106.4-full

### Please tell us more about your question or problem

We are using selfhosted renovate bot on our internal GH ent. org. with the following Workflow:
```yaml
---
name: renovate-bot

on:
  workflow_dispatch:
  merge_group:
  issues:
  push:
    branches:
      - main
  schedule:
    - cron: "0 2 * * 0-6"  # Everyday at 02:00

permissions:
  contents: write
  issues: write
  pull-requests: write
 
jobs:
  renovate:
    name: Renovate
    runs-on: selfhostedrunners
    if: ${{ github.actor != 'teamx-authentication[bot]' }}
    steps:
      - name: Checkout
        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332  # v4.1.7

      - name: Find Renovate config
        id: find-renovate-config
        run: |
          RENOVATE_CONFIG_FILE_PATH="$(find . -name renovate.json | head -n 1)"
          echo "RENOVATE_CONFIG_FILE_PATH=$RENOVATE_CONFIG_FILE_PATH" >> "$GITHUB_ENV"

      - uses: actions/create-github-app-token@31c86eb3b33c9b601a1f60f98dcbfd1d70f379b4  # v1.10.3
        id: generate-app-token
        with:
          app-id: ${{ secrets.GHA_TEAMX_APP_ID }}
          private-key: ${{ secrets.GHA_TEAMX_APP_PRIVATE_KEY }}

      - name: Run Renovate
        uses: renovatebot/github-action@630a255a1f2f56c8d8ce160bed3e3ca577ca53e2 # v40.2.7
        with:
          renovate-image: docker.internalreg.com/tool/renovate
          token: ${{ steps.generate-app-token.outputs.token }}
          configurationFile: ${{ env.RENOVATE_CONFIG_FILE_PATH }}
        env:
          RENOVATE_REPOSITORIES: ${{ github.repository }}
          LOG_LEVEL: "debug"
```
So currently, all our Repos have this workflow, and each of our GH Teams has its App to run Renovate with (we want to ensure that Team A can not write to a Repo of Team B with the GH App). So, our GitHub Apps are installed only on the Repos which belong to a specific team.

Now, we also have Repos, in which we ship internal GitHub Actions for the whole company; these Repos do not belong to a specific team; they are owned and managed by the GitHub Org admins. 

The problem we now have is that Renovate can not see the tags of these internal actions because the app is not installed there, e.g., Renovate does not have read access to the actions repos (see logs below).

So we tried the following things to solve this:
## Idea 1: Grant the App read access to the actions repo
Well, you can't. You need to install the app on the actions repo, and with this, you inherit the app repository permissions to that repository. Which would result in all people having write access to all actions-* repos. Individual repository permissions can only be given to teams or users, not apps.

## Idea 2: Use a Service account to run Renovate
This way, we could give the service account read access to all the actions-* Repos, but we would need one service account per team. Otherwise, we will have one mighty service account that can write to every repo, which is a security issue. Having one service account per team would cause additional costs because every user needs a license.

## Idea 3: Make actions Repo public
It does not work; our repos can only have the type "internal" or "private". With both of these, Renovate can not see them.

Any other ideas or help working around or solving this issue would be much appreciated. 

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Datasource unknown error (repository=intorg/some-repo)
       "datasource": "github-tags",
       "packageName": "intorg/actions-kustomize-setup",
       "err": {
         "message": "Could not resolve to a Repository with the name 'intorg/actions-kustomize-setup'.",
         "stack": "Error: Could not resolve to a Repository with the name 'intorg/actions-kustomize-setup'.\n    at GithubGraphqlDatasourceFetcher.doRawQuery (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:136:21)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at GithubGraphqlDatasourceFetcher.doShrinkableQuery (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:201:20)\n    at GithubGraphqlDatasourceFetcher.doPaginatedFetch (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:250:27)\n    at GithubGraphqlDatasourceFetcher.doCachedQuery (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:290:7)\n    at GithubGraphqlDatasourceFetcher.getItems (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:328:17)\n    at Function.query (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:60:19)\n    at queryTags (/usr/local/renovate/lib/util/github/graphql/index.ts:15:15)\n    at GithubTagsDatasource.getReleases (/usr/local/renovate/lib/modules/datasource/github-tags/index.ts:78:24)\n    at getRegistryReleases (/usr/local/renovate/lib/modules/datasource/index.ts:84:15)\n    at huntRegistries (/usr/local/renovate/lib/modules/datasource/index.ts:136:13)\n    at fetchReleases (/usr/local/renovate/lib/modules/datasource/index.ts:309:15)\n    at lookupUpdates (/usr/local/renovate/lib/workers/repository/process/lookup/index.ts:137:56)\n    at Function.wrap (/usr/local/renovate/lib/util/stats.ts:38:20)\n    at fetchDepUpdates (/usr/local/renovate/lib/workers/repository/process/fetch.ts:58:42)\n    at /usr/local/renovate/lib/workers/repository/process/fetch.ts:104:23\n    at /usr/local/renovate/node_modules/.pnpm/p-map@4.0.0/node_modules/p-map/index.js:57:22"
       }
DEBUG: Failed to look up github-tags package intorg/actions-kustomize-setup (repository=intorg/some-repo, packageFile=.github/workflows/dispatch-deploy-pr.yml, dependency=intorg/actions-kustomize-setup)
 

```

</details>

**Comment by zisuu** [comment]
Any idea how to solve this?

**Reply by rarkins** [reply]
Create a PAT with read-only access to the actions repos, add it to renovate using hostRules
